lib/util/git/index.ts

```diff
@@ -636,21 +639,49 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
     return isModified;
   }
 
-  logger.debug('branch.isModified(): using git to calculate');
-
   await syncGit();
-  // Retrieve the author of the most recent commit
-  let lastAuthor: string | undefined;
+  // Retrieve the commit authors
+  let branchAuthors: string[] = [];
   try {
-    lastAuthor = (
-      await git.raw([
-        'log',
-        '-1',
-        '--pretty=format:%ae',
-        `origin/${branchName}`,
-        '--',
-      ])
-    ).trim();
+    if (baseBranch) {
+      logger.debug(
+        `branch.isModified(): using git to calculate authors between ${branchName} and ${baseBranch}`
+      );
+      branchAuthors = [
+        ...new Set(
+          (
+            await git.raw([
+              'log',
+              '--pretty=format:%ae',
+              `origin/${branchName}..origin/${baseBranch}`,
```

Comment:
The syntax here is the wrong way.
This check means "all from baseBranch except those on branchName".

So basically this would return all commits that are on the base but NOT on the renovate-bot.

Reply:
what would be the right command?

Reply:
@viceice Either switch the operands
```suggestion
              `origin/${baseBranch}..origin/${branchName}`,
```

Or use the explicit
```suggestion
              `origin/${branchName} ^origin/${baseBranch}`,
```

Both should show all commits on the branchName (e.g. renovate/dep-abc), but _without_ any commits on the baseBranch (e.g. main).

So if we were to say (I hope the mermaid rendering is supported in all browsers)
```mermaid
gitGraph
       commit id: "A"
       commit id: "B"
       commit id: "C"
       branch renovate/x
       checkout main
       commit id: "D"
       commit id: "E"
       checkout renovate/x
       commit id: "X"
```

Then baseBranch would be `main` and branchName would be `renovate/x`
We'd _probably_ only want to look at the commit `X`, but the way this MR was written would return the D and E

Then the result would be:
| command | result | |
| ---------| --------| ----- |
| `main..renovate/x` | `X` |  ✅   |
| `renovate/x..main` | `D, E` | ❌  |
| `renovate/x ^main` | `X` |  ✅ |
| `main ^renovate/x` | `D, E` | ❌  |

---
Of course, if there were some rebasing done in between, it would be a bit different.  
Let's say we rebased C, D, E on the main branch (if for whatever reason you were to rebase on main).
Or well, pretty much anything that rewrites history, like someone else force-pushing over someone else's changes.

```mermaid
gitGraph
       commit id: "A"
       commit id: "B"
       branch renovate/x
       checkout renovate/x
       commit id: "C"
       commit id: "X"
       checkout main
       commit id: "C - rebased"
       commit id: "D - rebased"
       commit id: "E - rebased"
```

Then the result would be:
| command | result |
| ---------| --------|
| `main..renovate/x` | `C, X` |
| `renovate/x..main` | `C - rebased, D - rebased, E - rebased` |
| `renovate/x ^main` | `C, X`  |
| `main ^renovate/x` | `C - rebased, D - rebased, E - rebased`  |

--- 
I don't know what the expected behaviour should be in the case the main was force-pushed, but you can at least use these tables as reference if you wanted this feature again :wink:

Reply:
rebase would be an issue now too. I'm not too worried about that. it shouldn't be a normal workflow with renovate. can we detect that?

Reply:
maybe I'll try to create this PR again with proper tests